### PR TITLE
[#10195] Separate keystore dir per multiacc

### PR DESCRIPTION
**Pull request by rasom**
#10195

Having all keys related to specific multiacc in a separate dir will simplify removing of the multiacc.
Also it will allow adding of the same account to different multiaccs.

depends on https://github.com/status-im/status-go/pull/1993

status: ready

### CODE REVIEW

**Comment by hesterbruikman** [comment]
fyi @johnlea-quiup as this relates to key management, specifically removing a 'multiaccount' (master keypair)

File: src/status_im/multiaccounts/create/core.cljs [hunk 1]
```diff
@@ -71,6 +71,12 @@
 (fx/defn create-multiaccount
   [{:keys [db]}]
   (let [{:keys [selected-id key-code]} (:intro-wizard db)
+        key-uid (reduce
```
Comment (yenda) [inline src/status_im/multiaccounts/create/core.cljs]: unless I'm misunderstanding, it seems like `(first (filter f coll))` would be more idiomatic here

Comment (rasom) [reply src/status_im/multiaccounts/create/core.cljs]: that would be `(:key-uid (first (filter f coll)))`, but why it would be more idiomatic?

Comment (rasom) [reply src/status_im/multiaccounts/create/core.cljs]: btw then probably `(:key-uid (some f coll))` would make it with less parentheses. Idk how idiomatic it is `¯\_(ツ)_/¯`

Comment (rasom) [reply src/status_im/multiaccounts/create/core.cljs]: oh well it would be just `(some f coll)`, that should be a winner

**Comment by oskarth** [comment]
Small request - could you please rebase to latest master so this commit https://github.com/status-im/status-react/commit/b5fda1234e5447cbc626059dfc78fcfbabc9e926 gets included? This means we aren't running E2E tests on eth.prod cluster anymore, and since this happens on every push, the impact is quite big. This ensures we have more accurate metrics going forward from Jul 1 onward, which would be awesome. See https://discuss.status.im/t/user-growth-and-retention/1782 for more

**Review by flexsurfer — APPROVED** [review]

**Comment by churik** [comment]
## Can't send transaction from account if it is added as separate multiaccount with keycard

**Steps:**
- user A: restore multiaccount A from seed phrase A (choose keycard Storage), login
- user B: create account > go to wallet > Add account > Enter seed phrase A > Confirm, log out 
- User A: login to multiaccount A
- User A: try to send transaction 

**Expected result:** can send Transaction
**Actual result:** 
IMAGE 2020-07-03 17:36:03
[Status-debug-logs.zip](https://github.com/status-im/status-react/files/4870433/Status-debug-logs.zip)

**Comment by rasom** [comment]
> user A: restore multiaccount A from seed phrase A (choose keycard Storage), login
> user A: go to wallet > Add account > Enter seed phrase B > Confirm, log out

@churik is this even possible?

**Comment by churik** [comment]
Previous issue is not reproducible with the same steps, but I've got another error:

**Steps:**
1) user A: restore multiaccount A from seed phrase A (choose keycard Storage), login
2) user B: create account > go to wallet > Add account > Enter seed phrase A > Confirm, log out 
3) User A: login to multiaccount A > send transaction > log out
4) User B: login to multiaccount B > try to send transaction from added on 2 step account 

**Expected result:** can send Transaction
**Actual result:**
IMAGE 2020-07-06 13:36:14

Logs: [Status-debug-logs.zip](https://github.com/status-im/status-react/files/4878614/Status-debug-logs.zip)

**Comment by churik** [comment]
@rasom 
### 'Cannot retrieve a valid key for a given multiaccount error' after upgrade with 2 multiaccounts 
**Steps:**
1) installed Status Pr for upgrade (used from https://github.com/status-im/status-react/pull/10686: https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-200706-103428-6df9a2-pr-universal.apk)
2) recover multiaccount A
3) log out
4) create multiaccount B
5) upgrade to  PR build (https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-200706-131420-e9de81-pr-universal.apk)
6) try to login to ordinary multiaccount

**Expected result:**
logged in successfully

**Actual result:** error `Cannot retrieve a valid key for a given account: no key for given address or file"`
If  multiaccount A is keycard -> after login to multiaccount B/logout/login to multiaccount A, you'll see wallet/profile name of ordinary multiaccount B inside multiaccount A
[Status-debug-logs.zip](https://github.com/status-im/status-react/files/4885176/Status-debug-logs.zip)

**Comment by churik** [comment]
Tested on Android 10:
- upgrade with several multiaccounts (one of them was keycard with several accounts inside)
- added account from seed phrase to ordinary account (already opened as keycard multiaccount)
- checked transactions from added accounts

Tested on IOS 13:
- "Account already exists" (when add account from seed phrase within same multiccount or adding the same account twice)
- upgrade with several multiaccounts + enabled "Save my password"
- added account from seed phrase to ordinary account (already opened as keycard multiaccount) and vise versa
- checked transactions from added accounts.

Looks good to me,  thank you for patience @rasom

**Pull request merged**